src/client/components/select/view.tsx

```diff
@@ -0,0 +1,125 @@
+import * as classNames from 'classnames'
+import { action } from 'mobx'
+import { observable } from 'mobx'
+import { observer } from 'mobx-react'
+import * as React from 'react'
+import { ReactNode } from 'react'
+import OutsideClickHandler from 'react-outside-click-handler'
+
+import { Dropdown } from '../dropdown/view'
+
+import DropdownIcon from './dropdown.svg'
+import * as style from './style.css'
+
+export interface SelectOption {
+  id: string | number
+  label: string
+}
+
+export type SelectProps = {
+  className?: string
+  dropdownMenuClassName?: string
+  placeholder: string
+  options: SelectOption[]
+  selectedOption?: SelectOption
+  icon?: ReactNode
+  empty?: ReactNode
+  onChange(option: SelectOption): void
+}
+
+enum KeyCode {
+  Escape = 27,
+}
+
+@observer
+export class Select extends React.Component<SelectProps> {
+  @observable
+  private isOpen: boolean = false
```

Comment:
Probably.

I couldn't think of a nice way to allow it to be controlled from outside (via a prop) while still being able to open/close it from inside, since changing the prop would violate the one-way data flow from parent to child.

Maybe there's a pattern I'm missing here.

Reply:
Noted.